core/src/Streamly/Data/Array.hs

```diff
@@ -166,3 +170,7 @@ read = reader
 {-# INLINE_NORMAL readRev #-}
 readRev :: (Monad m, Unbox a) => Unfold m (Array a) a
 readRev = readerRev
+
+chunksSplitOn :: (Monad m, Unbox a) =>
+    (a -> Bool) -> Array a -> Stream m (Array a)
+chunksSplitOn = A.splitOn
```

Comment:
This is not what I meant by chunksSplitOn. You can remove this from this PR. We can discuss what is the right thing and add it in another PR.